setup/advanced/ipv6.md

```diff
@@ -0,0 +1,44 @@
+# IPv6
+
+⚠️ IPv6 is an experimental Docker feature, no thorough test was done to ensure there is no IP/DNS leak with Gluetun configured with IPv6. Feel free to create an issue or pull request if you have some testing done and can confirm.
+
+## Setup
+
+1. On your Docker host, edit and create if needed `/etc/docker/daemon.json` with the following JSON key-value pairs:
+
+    ```json
+    {
+      "ipv6": true,
+      "fixed-cidr-v6": "fd00::/80",
+      "experimental": true,
+      "ip6tables": true
+    }
+    ```
+
+    [Reference: Docker documentation on IPv6](https://docs.docker.com/config/daemon/ipv6/#use-ipv6-for-the-default-bridge-network)
+
+    ⁉️ IP range change to unique local address following the note at button of the documentation.
+1. Restart the Docker daemon to reload its JSON configuration. Most Linux distributions use `sudo systemctl restart docker` to do this.
+1. Edit your Gluetun `docker-compose.yml` and add the `sysctls` section and modify `WIREGUARD_ADDRESSES` to have both an IPv4 and an IPv6 address:
+
+    ```yaml
+    services:
+      gluetun:
+        image: qmcgaw/gluetun
+        container_name: gluetun
+        # ...
+        environment:
+          WIREGUARD_ADDRESSES=xxx.xxx.xxx.xxx/32,fd7d:.............../128
+        sysctls:
+          - net.ipv6.conf.all.disable_ipv6=0
+    ```
+
+1. Test your setup:
+    1. Launch your docker-compose stack
+    1. Run:
+
+        ```sh
+        sudo docker run --rm --network=container:gluetun alpine:3.18 sh -c "apk add curl && curl -6 --silent https://ipv6.ipleak.net/json/"
```

Comment:
I changed this to directly curl ipleak over ipv6, this should work nicely (can you try it please 🙏 I don't have ipv6 over here, shame on me 😢

Reply:
Container do change depending on the directory name of the directory of docker-compose.yml

![](https://i.imgur.com/OqBp157.png)

When I put docker-compose.yml in "gluetun" directory, the name is "gluetun-gluetun-1".

When I put docker-compose.yml in "ipv6" directory, the name is "ipv6-gluetun-1".

Reply:
I am not sure if the changing container name has anything to do with me using `sudo docker-compose up`

Reply:
I can make sure gluetun's container is "gluetun" by using `sudo docker-compose run --rm --name gluetun gluetun`

Reply:
Awesome, what matters is that curl command works well 💯

You can just set `name: gluetun` in your docker-compose.yml to always have the same name for the container. Otherwise yeah it's directory name + compose spec definition name + index.